libs/application/templates/energy-funds/src/fields/SelectVehicle/VehicleSelectField.tsx

```diff
@@ -0,0 +1,159 @@
+import { FieldBaseProps, Option } from '@island.is/application/types'
+import { useLocale } from '@island.is/localization'
+import { FC, useCallback, useState } from 'react'
+import {
+  ActionCard,
+  AlertMessage,
+  Box,
+  SkeletonLoader,
+} from '@island.is/island-ui/core'
+
+import { SelectController } from '@island.is/shared/form-fields'
+import { useFormContext } from 'react-hook-form'
+import { getValueViaPath } from '@island.is/application/core'
+import { information } from '../../lib/messages/information'
+import { VehiclesCurrentVehicle } from '../../shared/types'
+import { useLazyVehicleDetails } from '../../hooks/useLazyVehicleQuery'
+import format from 'date-fns/format'
+
+interface VehicleSearchFieldProps {
+  currentVehicleList: VehiclesCurrentVehicle[]
+}
+
+export const VehicleSelectField: FC<
```

Comment:
I would like to revisit the buildSelect fields once this has been released just to see what we are missing in the base field to make this case work.